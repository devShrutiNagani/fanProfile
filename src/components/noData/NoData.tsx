import {View, Text, Image} from 'react-native';
import React from 'react';
import createStyles from './Styles';
import {NoDataProps} from 'constant/interface/custom/Custom';

const NoData: React.FC<NoDataProps> = ({
  containerStyle,
  image,
  imageStyle,
  title,
  titleStyle,
  message,
  messageStyle,
}) => {
  const styles = createStyles();

  return (
    <View style={[styles.container, containerStyle]}>
      <Image style={[styles.image, imageStyle]} source={image} />
      <View style={styles.textView}>
        <Text style={[styles.title, titleStyle]}>{title}</Text>
        <Text style={[styles.message, messageStyle]}>{message}</Text>
      </View>
    </View>
  );
};

export default NoData;
