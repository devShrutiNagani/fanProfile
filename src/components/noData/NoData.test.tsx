import React from 'react';
import {render} from '@testing-library/react-native';
import NoData from './NoData';

describe('NoData Component', () => {
  it('renders title and message correctly', () => {
    const {getByText} = render(
      <NoData title="No Data Available" message="Please try again later" />,
    );

    expect(getByText('No Data Available')).toBeTruthy();
    expect(getByText('Please try again later')).toBeTruthy();
  });

  it('renders the image when provided', () => {
    const {getByTestId} = render(<NoData image={{uri: 'test-image'}} />);

    expect(getByTestId('no-data-image')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const tree = render(
      <NoData
        title="Snapshot Title"
        message="Snapshot Message"
        image={{uri: 'test-image'}}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
