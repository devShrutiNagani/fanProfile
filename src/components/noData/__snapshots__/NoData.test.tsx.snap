// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoData Component matches snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Image
    source={
      {
        "uri": "test-image",
      }
    }
    style={
      [
        {
          "height": 294,
          "resizeMode": "contain",
          "width": 589,
        },
        undefined,
      ]
    }
    testID="no-data-image"
  />
  <View
    style={
      {
        "alignItems": "center",
        "marginHorizontal": 39,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#1D1D1D",
            "fontFamily": "Montserrat-SemiBold",
            "fontSize": 27,
            "includeFontPadding": false,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Snapshot Title
    </Text>
    <Text
      style={
        [
          {
            "color": "#1D1D1D",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 24,
            "includeFontPadding": false,
            "marginTop": 10,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Snapshot Message
    </Text>
  </View>
</View>
`;
